PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,12 +1,42 @@
-<!-- Your title above should be a short description of what
-was changed. Do not include the issue number in the title. -->
+<!--
+Thank you so much for your PR!  To help us review your contribution, please
+consider the following points:
+
+- Your title above should be a short description of what was changed. Do not
+  include the issue number in the title.
+
+- Development information is available at
+  https://github.com/sympy/sympy/wiki/Development-workflow.
+
+- Do not create the PR out of master, but out of a separate branch.
+
+- Try to keep the number of commits to a minimum. Use rebase and
+  force pushing in git to avoid adding unnecessary commits.
+
+- If this is your first commit, see:
+  https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file
+
+- If you are contributing fixes to docstrings, please pay attention to
+  https://docs.sympy.org/dev/guides/contributing/documentation-style-guide.html.
+  In particular, note the difference between using single backquotes,
+  double backquotes, and asterisks in the markup.
+
+- Please include tests of any new feature or bug fix. The code in the tests
+  should not execute correctly in a version of SymPy without your code.
+
+Note that all reviewing is done an a volunteer basis, so it can sometimes take
+time before you get feedback. If you have not heard back in two weeks, feel
+free to ping @sympy/developers-with-push-access-to-everything
+Do not ping individual persons to get a review, nor request reviews.
+Also note that any reviewer will be notified of changes in your PR, so there
+is no need to ping after a change.
+-->
 
 #### References to other Issues or PRs
 <!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
```

Comment:
Can we add some text here stating that if an issue does not already exist then it is not necessary to open one. That has confused a lot of new contributors.